### ISSUE TITLE
Flatpak Aegisub Cannot Access Files Outside of Home Directory

### ISSUE THREAD

**Issue by gourdcaptain** [flathub/org.aegisub.Aegisub#10]
I keep video files I edit subtitles on with Aegisub stored on a NAS I mount under `/mnt/nas` due to the large and bulky size of the collection relative to the internal disk space of the systems in question. The current version of the Aegisub flatpak is limited to folders in the home directory, so any sources of files that are on removable drives or other filesystems are just invisible inside the Flatpak. I knew enough to use Flatseal to add the filesystem, but many users would be stumped and stuck at this point. 

I think there's ways for programs designed for portal integration to get around this, but I'm guessing Aegisub as an older program largely in maintenance mode doesn't support this?

**Comment by zefr0x** [comment]
> I think there's ways for programs designed for portal integration to get around this, but I'm guessing Aegisub as an older program largely in maintenance mode doesn't support this?

`Aegisub` uses `wxWidgets` which by itself could be configured in build-time to use `GTK3` (The option that is used for this flatpak build).

In `GTK4` it would be enabled by default, but in `GTK3` you will need to set `GTK_USE_PORTAL=1` in your environment.

Any way, for it to work `wxWidgets` must use `GtkFileChooserNative` as the file picker dialog, which is yet an open issue: https://github.com/wxWidgets/wxWidgets/issues/24486


When this issue get resolved I will set `GTK_USE_PORTAL=1` in this flatpak environment, so users don't need to do any thing.

~I don't think that we can drop the current file-system permissions (`--filesystem=home`) even after that, since we still can't open files from our file-browser without it, unless there was a portal to give permissions for files from other applications.~

**Comment by zefr0x** [comment]
https://github.com/wxWidgets/wxWidgets/pull/25152 has been merged into master, but wasn't released yet.

When it get released `git tag --contains 071b4dd0f105e6d6ecd210190ce65bf41d6f1e2a` should give a version tag.

**Comment by zefr0x** [comment]
It has been released in wxWidgets `v3.3.0`, pending upstream Aegisub to release with breaking changes fixed.
